src/core/lib/mman/mman.cc

```diff
@@ -113,12 +113,9 @@
       const DWORD protect = __map_mmap_prot_page(prot, flags);
       const DWORD desiredAccess = __map_mmap_prot_file(prot, flags);
 
-      const off_t maxSize = off + (off_t)len;
-
-      const DWORD dwMaxSizeLow = (sizeof(off_t) <= sizeof(DWORD)) ?
-                      (DWORD)maxSize : (DWORD)(maxSize & 0xFFFFFFFFL);
-      const DWORD dwMaxSizeHigh = (sizeof(off_t) <= sizeof(DWORD)) ?
-                      (DWORD)0 : (DWORD)((maxSize >> 32) & 0xFFFFFFFFL);
+      const size_t maxSize = (size_t)off + len;
+      const DWORD dwMaxSizeLow = (DWORD)(maxSize & 0xFFFFFFFFL);
+      const DWORD dwMaxSizeHigh = (DWORD)((maxSize >> 32) & 0xFFFFFFFFL);
```

Comment:
We should probably have a `static_assert` about the size of `DWORD` somewhere

Reply:
I can add it, but as far as I can see it is 32 bits on all the Windows systems.

Reply:
A static assert will make this knowledge explicit. Plus, if the new version of windows comes out with 64 bit dwords, then  we'll at least be able to catch it.

Reply:
Sure, will add it. But my bet is that when a new version of Windows comes out, the code won't even compile :)